Frameworks/CoreText/DWriteWrapper.mm

```diff
@@ -47,12 +47,56 @@
     return s_dwriteFactory;
 }
 
+template <typename TElement>
+bool __CloneArray(_In_reads_opt_(count) TElement const* source,
```

Comment:
This seems like the sort of thing that will have an STL solution already...

does std::copy suit your needs?